### ISSUE TITLE
Dedicated documentation on Alignment?

### ISSUE THREAD

**Issue by TomNicholas** [pydata/xarray#9500]
### What is your issue?

I'm writing more documentation for DataTree, and was looking for an in-depth explanation of what Alignment actually is so I can refer to it, but I can't really find one?

All I can find 
The concept of alignment is referred to in multiple places, e.g.
- There's is a brief definition of alignment in the [terminology page](https://docs.xarray.dev/en/latest/user-guide/terminology.html#term-Aligning),
- There's [some explanation](https://docs.xarray.dev/en/stable/user-guide/indexing.html#align-and-reindex) in the context of using `.reindex()`,
- Some more in the context of [automatic alignment](https://docs.xarray.dev/en/latest/user-guide/computation.html#automatic-alignment) during [broadcasting](https://docs.xarray.dev/en/latest/user-guide/computation.html#broadcasting-by-dimension-name),
- Alignment of indexes is referred to in the documentation of `xr.merge` on the [Combining page](https://docs.xarray.dev/en/latest/user-guide/combining.html#merge)
- The closest thing to dedicated documentation I've found is in the explanation of `Dataset.update()` [on the data structures page](https://docs.xarray.dev/en/latest/user-guide/combining.html#update).

But there doesn't appear to be a dedicated section on Alignment anywhere in the narrative docs (or even on the xarray tutorial site). I can't find any explanation of what it means for two `Datasets` not to align, for example.

This seems like a very fundamental concept to not have dedicated documentation. Maybe I've just missed it?

cc @max-sixty @shoyer

**Comment by dcherian** [comment]
Yeah its not great. I wrote some tutorial material here: https://tutorial.xarray.dev/fundamentals/02.3_aligning_data_objects.html

**Comment by TomNicholas** [comment]
> I wrote some tutorial material here:

I was looking for that one! The content on that really fills a much-needed gap (and is really good). I think a dedicated page that explains what Broadcasting and Alignment are is what we are missing from the main docs.

**Comment by benbovy** [comment]
In general I find [https://tutorial.xarray.dev/](https://tutorial.xarray.dev/) really well done, it would definitely deserve more exposure / discoverability / integration with the main docs!
